docs/Development/Release-Process.md

```diff
@@ -0,0 +1,62 @@
+# Release Process
+
+From early on, Diamond was run as a master is stable release style. The belief was that master should always be stable and ready to deploy to production. This lead to the odd release system we currently have.
+
+## Version String
+
+The current version is generated by the bash script at https://github.com/python-diamond/Diamond/blob/master/version.sh
+
+The major and minor version numbers are generated based on the latest annotated tag.
+
+Cutting a new major/minor version is done via:
+
+```shell
+git tag -a v4.1 -m "foobar"
+git push origin --tags
+```
+
+From this point forward, the makefile will generate a version starting with 4.1
```

Comment:
Sorry, I wasn't clear. If we split up all collectors/handlers into their own repos, what is a minor point release of the core at that point? Or do we bring them all together to package them so it's purely just a code artifact that they're split up?